drivers/hmis/app/graphql/mutations/submit_form.rb

```diff
@@ -0,0 +1,86 @@
+module Mutations
+  class SubmitForm < BaseMutation
+    description 'Submit a form to create/update HUD record(s)'
+
+    argument :input, Types::HmisSchema::FormInput, required: true
+
+    field :record, Types::HmisSchema::SubmitFormResult, null: true
+
+    def resolve(input:)
+      errors = HmisErrors::Errors.new
+
+      # Look up form definition
+      definition = Hmis::Form::Definition.find_by(id: input.form_definition_id)
+      errors.add :form_definition, :required unless definition.present?
+      return { errors: errors } if errors.any?
+
+      # Determine record class
+      klass = definition.record_class_name&.constantize
+      errors.add :form_definition, :invalid unless klass.present?
+      return { errors: errors } if errors.any?
+
+      # Find or create record
+      hud_user = Hmis::Hud::User.from_user(current_user)
+      if input.record_id.present?
+        record = klass.viewable_by(current_user).find_by(id: input.record_id)
+        record&.assign_attributes(user: hud_user)
+      else
+        record = klass.new(user: hud_user, data_source_id: hud_user.data_source_id)
+        input.apply_related_ids(record)
+      end
+
+      errors.add :record, :not_found unless record.present?
+      return { errors: errors } if errors.any?
+
+      # Check permission
+      has_permission = if record.is_a?(Hmis::Hud::Organization) || record.is_a?(Hmis::Hud::Client)
```

Comment:
Why only organizations and clients?  Are there no direct permissions for projects?  (Maybe a note here about why these two are different.)

Reply:
Alternatively, maybe `permissions_for?` should handle the special cases?